test/test_torch.py

```diff
@@ -4224,11 +4224,13 @@ def test_serialization(self):
         b += [(t1.storage(), t1.storage(), t2.storage())]
         b += [a[0].storage()[0:2]]
         for use_name in (False, True):
-            with tempfile.NamedTemporaryFile() as f:
-                handle = f if not use_name else f.name
-                torch.save(b, handle)
-                f.seek(0)
-                c = torch.load(handle)
+            f = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
In this case we are not able to use context manager to manage the lifecycle of the file, because Windows doesn't support opening an already opened file (which we are doing in `torch.save(...)` when only file name is passed to `handle`, see https://github.com/pytorch/pytorch/blob/master/torch/serialization.py#L114)

Reply:
Just a note here: This one would left the file undeleted.

Reply:
A better fix is to skip test when use_name=True.

Reply:
I believe `os.unlink(f.name)` below should delete the file?

Reply:
No, it won't. It will be same error.

Reply:
So if you don’t use t as a context manager then the file doesn’t even get opened? It seems that it’s not the case since you’re sill calling close(), so I’m not sure what’s the problem.

Reply:
The real problem is hard to describe. It's a flag `FILE_FLAG_DELETE_ON_CLOSE` in the function `os.read` used by the package `tempfile` that prevents the file from opening the second time. To resolve this, there're two ways:
1. Introduce a Windows version of `os.read`.
2. Skip this test since loading from the handle can also test the functionality of serialization.